### PR TITLE
Lift criterion

### PR DESCRIPTION
**Pull request by apetresc**
This reapplies the changes in [bjlkeng/scikit-learn@lift_criterion](https://github.com/bjlkeng/scikit-learn/tree/lift_criterion), applied to the latest upstream `v0.17.1-1` tag and tweaked to make sure tests pass.

Unfortunately the generated `.c` files are still here because they weren't removed in upstream until after 0.17.1, hopefully we can get rid of them after the next release.

All unit tests pass.

### CODE REVIEW

File: sklearn/tree/tests/test_tree.py [hunk 1]
```diff
@@ -1324,3 +1326,103 @@ def test_presort_sparse():
...
+    # Check classification on a toy dataset.
+    clf = DecisionTreeClassifier(criterion='lift', random_state=0)
+    clf.fit(X, y)
+
+    result = clf.predict_proba(T)
+    assert_array_equal(result, true_result,
+                       "Failed with prediction: {}".format(result))
+
+
+def test_classification_lift_toy4():
```
Comment (bjlkeng) [inline sklearn/tree/tests/test_tree.py:1398]: Were you able to re-run unit tests to see if we pass all of them?

**Review by bjlkeng — COMMENTED** [review]
Just one question about unit tests, otherwise looks right.

**Review by bjlkeng — COMMENTED** [review]
Just a question about unit tests, otherwise looks good.

File: sklearn/tree/tests/test_tree.py (hunk 1 above)
Comment (apetresc) [reply sklearn/tree/tests/test_tree.py:1398]: Yes, and they did (after I removed some graphviz dot files you were generating in the new unit tests)

**Pull request merged**